### PR TITLE
Balance game with changable bullet speed during boss phase

### LINKED ISSUE #530: Game Balance
# **Description**
**Task:** Balance the game 
Game is working well but values need to be tweaked.

Ensure the game feels fair for the player and still challenging enough. 
Pets
Enemies 
Bosses 
Enemy Spawning 
Weapons

**Completion Deadline:** Oct 16

# **Documentation**
- Game Balance page 
# **Member**
- Name (GitHub Username) (Slack Handle)
- Varun Sinha @vsinh14

### PR DESCRIPTION
**Pull request by justarandomnameduh**
# Description

Add a new bullet speed coefficient to `RangeAttackComponent` and change it during `BulletStormTask` duration. The bullet speed is changed within `RangeAttackComponent` and `BossRangeAttackComponent`.

Change the projectiles.json for corresponding change in new speed coefficient.

Fixes #530

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Test manually with the change of projectile speed during boss phase.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: source/core/src/main/com/csse3200/game/components/npc/attack/BossRangeAttackComponent.java [hunk 1]
```diff
@@ -100,6 +104,9 @@ private void shoot() {
 
     private void singleShoot(Vector2 spawnLocation, Vector2 movingDirection) {
         Entity projectile = projectileFactory.create(getProjectileNames()[getAnimationID()], movingDirection, spawnLocation);
+        ProjectileAttackComponent projectileAttackComponent = projectile.getComponent(ProjectileAttackComponent.class);
+        Vector2 v = projectileAttackComponent.getSpeed();
+        projectileAttackComponent.setSpeed(new Vector2(v.x * speedCoefficient, v.y * speedCoefficient));
```
Comment (IanP96) [inline source/core/src/main/com/csse3200/game/components/npc/attack/BossRangeAttackComponent.java:109]: can use Vector.scl which should also work but no issues with this

Comment (vsinh14) [reply source/core/src/main/com/csse3200/game/components/npc/attack/BossRangeAttackComponent.java:109]: So you good to approve it?

**Review by mblauberg — APPROVED** [review]
worked well for me when testing

**Review by Aryan-S-Gupta — APPROVED** [review]
LGTM

**Review by vsinh14 — APPROVED** [review]
Changed the frequency of base projectile firing by the boss but looks good otherwise

**Review by CameronBadman — APPROVED** [review]
LGTM

**Review by CameronBadman — APPROVED** [review]
LGTM

**Pull request merged**